provider/datagen/src/bin/datagen.rs

```diff
@@ -106,11 +107,6 @@ fn main() -> eyre::Result<()> {
                 .help("Whether to optimize Unicode property data structures for size (\"small\") or speed (\"fast\")")
                 .default_value("small"),
         )
-        .arg(
-            Arg::with_name("INPUT_FROM_TESTDATA")
```

Comment:
Question: What does this PR achieve?

A long time ago, we used to have two separate binary tools that shared a library API, much like you're proposing here. I merged them into one tool because I found that I often wanted to use the flags from the more full-featured tool in the one that generated testdata. This PR basically brings us back to that state, where `icu4x-testdata-datagen` is not configurable. The status quo has been working really well and I don't see why we want to change it again.